remote-config/src/fetch/fetcher.rs

```diff
@@ -0,0 +1,316 @@
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::{Arc, Mutex, MutexGuard};
+use base64::Engine;
+use hyper::http::uri::{PathAndQuery, Scheme};
+use hyper::{Body, Client, StatusCode};
+use sha2::{Digest, Sha256, Sha512};
+use tracing::{debug, trace, warn};
+use datadog_trace_protobuf::remoteconfig::{ClientGetConfigsRequest, ClientGetConfigsResponse, ClientState, ClientTracer, ConfigState, TargetFileHash, TargetFileMeta};
+use ddcommon::{connector, Endpoint};
+use crate::{RemoteConfigPath, Target};
+use crate::targets::TargetsList;
+
+const PROD_INTAKE_SUBDOMAIN: &str = "config";
+
+/// Manages files.
```

Comment:
Talking about files when referring to configurations can be misleading. It's a ConfigurationStore, replace the naming from file to config please

Reply:
I took the name from `message File` from the proto definition. It's literally the contents of these `File` messages, which are meant processed here.